### DISCUSSION TITLE
멀티플레이어 실행 시 다양한 문제 발생(아바타 복제/모바일 테스트 시 한사람만 들어가지는 현상)

### DISCUSSION THREAD

**Discussion by IlsYoungOG** [naverz/zepeto-studio-kor#854] [5. Q&A]
안녕하세요. IlsYoungOG입니다. 

제페토 멀티플레이 강의를 보고 진행을 하였는데요, 유니티 테스트를 하면 멀티플레이어가 성공했다고 알림이 뜨면서 프로젝트가 실행이 됩니다.
그런데 2가지의 문제가 있는데요

<img width="1792" alt="스크린샷 2022-06-08 오후 9 59 55" src="https://user-images.githubusercontent.com/107122738/172623851-fc9734e2-4a41-48c3-9448-27fb22e28f1a.png">

첫 번째는 아바타가 복제됩니다. 한 아바타는 가만히 있고 한 아바타는 컨트롤을 하면 움직입니다. 

<img width="1784" alt="스크린샷 2022-06-08 오후 10 01 24" src="https://user-images.githubusercontent.com/107122738/172623895-93018f9f-d984-44b0-9ac4-d017f5f6cf19.png">

두 번째, 다른 계정으로 QR코드로 접속하면 제작한 계정으로만 들어가집니다. 또한 분명히 알림에는 성공이라고 뜨는데 동시접속도 되지 않습니다.

<img width="1792" alt="스크린샷 2022-06-08 오후 10 02 38" src="https://user-images.githubusercontent.com/107122738/172624019-ed982be2-719d-4950-97e3-e9dcf77d82d3.png">

<img width="1792" alt="스크린샷 2022-06-08 오후 10 03 26" src="https://user-images.githubusercontent.com/107122738/172624083-5150443a-e19a-4903-9346-d4fd605081a1.png">

<img width="1792" alt="스크린샷 2022-06-08 오후 10 03 36" src="https://user-images.githubusercontent.com/107122738/172624143-7555fc0e-8bfd-453f-aac9-216872e48478.png">

세 번째, 이벤트 오류입니다. (47번째 줄에서 add_OnChange라는 이벤트를 등록하려고 했는데 정의가 되지 않았다고 합니다.)

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, hierarchy에 있는 zepetocontroller에 혹시 캐릭터를 생성하는 코드가 들어있지는 않나요?
그리고 오류가 뜬 스크립트를 클릭하면 인스펙터에 Open Compiled Script가 뜹니다. 
타입스크립트를 컴파일하면 js형태가 되는데 이렇게 열어서 확인한 스크립트의 47번째 줄을 살펴보시기를 추천 드립니다.
![스크린샷 2022-06-09 오전 10 54 18](https://user-images.githubusercontent.com/100118650/172747502-1bc4fb7a-4524-467c-8311-64dd9886de15.png)
